tests/scripts/thread-cert/Cert_5_3_07_DuplicateAddress.py

```diff
@@ -31,96 +31,120 @@
+        self.nodes[SED1].set_panid(0xface)
+        self.nodes[SED1].set_mode('rsn')
+        self.nodes[SED1].add_whitelist(self.nodes[ROUTER2].get_addr64())
+        self.nodes[SED1].enable_whitelist()
 
-        self.nodes[ED3].set_panid(0xface)
-        self.nodes[ED3].set_mode('rsn')
-        self.nodes[ED3].add_whitelist(self.nodes[LEADER].get_addr64())
-        self.nodes[ED3].enable_whitelist()
+        self.nodes[MED3].set_panid(0xface)
+        self.nodes[MED3].set_mode('rsn')
+        self.nodes[MED3].add_whitelist(self.nodes[DUT_LEADER].get_addr64())
+        self.nodes[MED3].enable_whitelist()
+
+        self.sniffer = config.create_default_thread_sniffer()
+        self.sniffer.start()
 
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
-        self.nodes[LEADER].start()
-        self.nodes[LEADER].set_state('leader')
-        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+        # 1
+        self.nodes[DUT_LEADER].start()
+        self.nodes[DUT_LEADER].set_state('leader')
+        self.assertEqual(self.nodes[DUT_LEADER].get_state(), 'leader')
 
-        self.nodes[ROUTER1].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ROUTER1].get_state(), 'router')
+        for DEV in range(ROUTER1, MED3 + 1):
+            self.nodes[DEV].start()
 
-        self.nodes[ROUTER2].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[ROUTER2].get_state(), 'router')
 
-        self.nodes[ED1].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED1].get_state(), 'child')
+        for ROUTER in [ROUTER1, ROUTER2]:
+            self.assertEqual(self.nodes[ROUTER].get_state(), 'router')
 
-        self.nodes[ED2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED2].get_state(), 'child')
+        for ED in MTDS:
+            self.assertEqual(self.nodes[ED].get_state(), 'child')
 
-        self.nodes[ED3].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED3].get_state(), 'child')
+        # 2
+        leader_messages = self.sniffer.get_messages_sent_by(DUT_LEADER)
+        msg = leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+        command.check_mle_advertisement(msg)
 
+        # 3
         self.nodes[ROUTER2].add_prefix('2001:2:0:1::/64', 'paros')
         self.nodes[ROUTER2].register_netdata()
+        # Wait for ROUTER2 to configure slaac address.
+        time.sleep(25)
 
-        self.nodes[ED1].add_ipaddr('2001:2:0:1::1')
-        self.nodes[ED2].add_ipaddr('2001:2:0:1::1')
-        time.sleep(5)
+        self.nodes[MED1].add_ipaddr('2001:2:0:1::1234')
+        self.nodes[SED1].add_ipaddr('2001:2:0:1::1234')
+        # Wait for MED1/SED1 to configure slaac address.
+        time.sleep(10)
+        self.sniffer.get_messages_sent_by(DUT_LEADER)
```

Comment:
Move Line 132 after below Line 134 and add below comment as in other cases:
`# Flush the message queue to avoid possible impact on follow-up verification.`